pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -190,15 +195,38 @@ func (a *AmbientIndexImpl) All() []*model.AddressInfo {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 	res := make([]*model.AddressInfo, 0, len(a.byUID)+len(a.serviceByNamespacedHostname))
+
+	addedWEAddr := map[networkAddress]*model.AddressInfo{}
+	allNetworkAddresses := sets.New[networkAddress]()
+
 	for _, wl := range a.byUID {
-		res = append(res, workloadToAddressInfo(wl.Workload))
+		addressInfo := workloadToAddressInfo(wl.Workload)
+		networkAddr := networkAddressFromWorkload(wl)
+		// For WorkloadEntry we need to check whether a workload with similar network/address exists.
+		// WorkloadEntry may have a single address.
```

Comment:
Is this because recently @stevenctl made wle address optional?

Reply:
no this part is unrelated to that.

the problem arises when a WE has the same workload ip defined as a pod in the mesh. network+ip are unique in ambient (for on-demand lookups) so the intent was to check for any invalid configuration given these constraints